203 switch to openmpi for linux packages

bruno-at-orange opened the pull request:
Switch to OpenMPI for rpm and deb packages. The MPI implementation is now a parameter of the actions `pack-debian` and `pack-rpm.` The packages and binaries names contain the name of the mpi implementation (openmpi or mpich).

The `run-unit-test` action works indifferently with openMPI or mpich by using ctest (close #23).
However, the `run-standard-test` action works only with binaries built with openmpi.

close #203 
may close the issue #164 as we use openMPI as default mpi implementation

Extra features (unrelated with openmpi):
- Add environment variable to launch a file server process on single host 
- close #216 
- close #207
- cherry pick LearningTestTool from dev

bruno-at-orange committed:
switch to OpenMPI

Update version to 10.2.2-b.0

Build new docker images with MPICH and OpenMPI.

Generates DEB and RPM packages for multiple mpi implementations (mpich and openMPI)
- Add mpi implemntation discovering in cmake (not working in conda)
- Add a suffix to package names corresponding to the mpi implementation (as it advised in fedora packaging guidline)
- Add mpi implementation name in packages names
- packages khiops-core-openmpi with a version >> 10.2.1 replace khiops-core-10.2.1 without trouble with apt and dnf
- Add mpich vs openMPI option in debian and rpm workflows (openMPI is the default)
- Use ctest to run unit tests (issue #23)

Cherry pick LearningTestTool and test hierarchy from dev

Handle binary renaming in LearningTestTool
- Switch to openMPI in kht_test and standard tests
- The kht_test script handles the new naming of the parallel binaries: MODL_openmpi or MODL_mpich or MODL (the last one has priority)

Conda:
- Fix mpich build
- Remove the use of the obsolete cmake option MPI_SUFFIX in conda build

Fix DEB and RPM version renaming in pack-debian and pack-rpm

bruno-at-orange committed:
Change exit status on MODL and MODL_Coclustering

return 0 on success, 1 on fatal error
If there is any error or warning in the log file, the exit code is 0
fix issue https://github.com/KhiopsML/khiops/issues/216

bruno-at-orange committed:
Add -s flag and remove messages about drivers on stdout

cherry pick from dev
fix issue https://github.com/KhiopsML/khiops/issues/207

bruno-at-orange committed:
Add environment variable to launch a file server process on single host

When the environment variable KhiopsFileServerActivated is set to true, an MPI file server process
is launched on a single machine. These process are used on clusters to exchange temporary files.
This feature allows to test the file server on a common laptop.

folmos-at-orange reviewed (changes requested):
Mostly cosmetic stuff.

One question: It is possible to have just one binary which can be linked to both OpenMPI and MPICH ?

_Edited_


And another two questions: 
- Did we test for performance degradations ?
- It is pertinent to migrate to openmpi for conda packages ?

bruno-at-orange committed:
remove post-installations script from khiops packages

Java configuration is in khiops-env in the package khiops-core. It is not updated by khiops package anymore.
The trick is that this configuration is triggered by the presence of khiops.jar

On the docker files, We pin the jdk version to the smallest one on each distro (it matches most of the time the default-jdk package)
The idea is to build norm.jar with the oldest jdk to have a jar compatible with all jre. Ideally we should build with jdk 1.8 on all
distros but this package is not available everywhere.

cherry pick from dev PR #245

folmos-at-orange reviewed (approved):
I think the version should be changed. See comment.

If it is possible, could we have a "no suffix" generation even if MPI is available ? In conda we do not need the suffixes, and it simplifies the lookup for the executable. At the moment I just rename it after the build.
